### PR TITLE
Introduce "Span" class for upcoming *.d.ts generator

### PR DESCRIPTION
**Pull request by pgonzal**
This is the first set of changes being peeled off from the *.d.ts prototype.  The Span class is a helper that allows us to rewrite TypeScript source code without having to understand every aspect of the AST structure.

### CODE REVIEW

**Comment by pgonzal** [comment]
@RyanCavanaugh FYI

File: libraries/api-extractor/src/IndentedWriter.ts [hunk 1]
```diff
@@ -44,8 +65,8 @@ export default class IndentedWriter {
    * Each call to IndentedWriter.increaseIndent() must be followed by a
    * corresponding call to IndentedWriter.decreaseIndent().
    */
-  public increaseIndent(prefix: string = '  '): void {
-    this._indentStack.push(prefix);
+  public increaseIndent(): void {
+    this._indentStack.push(this.spacing);
     this._updateIndentText();
   }
```
Comment (nickpape-msft) [inline libraries/api-extractor/src/IndentedWriter.ts]: I thought we already had a class for this?

Comment (pgonzal) [inline libraries/api-extractor/src/IndentedWriter.ts]: ?  The IndentedWriter is not new in this PR

File: libraries/api-extractor/testInputs/SpanTest1-Expected.txt [hunk 2]
```diff
@@ -0,0 +1,115 @@
+SourceFile:
```
Comment (pgonzal) [inline libraries/api-extractor/testInputs/SpanTest1-Expected.txt]: (This is the output of Span.getDump())

File: libraries/api-extractor/src/IndentedWriter.ts (hunk 1 above)
Comment (nickpape-msft) [inline libraries/api-extractor/src/IndentedWriter.ts]: Nevermind, we do have a class, but it is checked into a private project.

---
In reply to: [132347984](https://github.com/Microsoft/web-build-tools/pull/304#discussion_r132347984) [](ancestors = 132347984)

**Comment by nickpape-msft** [comment]
:shipit:

**Comment by pgonzal** [comment]
:shipit:

**Pull request merged**